stl/inc/concepts

```diff
@@ -201,41 +195,21 @@ concept copy_constructible = move_constructible<_Ty>
     && constructible_from<_Ty, const _Ty&> && convertible_to<const _Ty&, _Ty>
     && constructible_from<_Ty, const _Ty> && convertible_to<const _Ty, _Ty>;
 
-// CONCEPT movable
+// CONCEPT _Boolean_testable
 template <class _Ty>
-concept movable = is_object_v<_Ty> && move_constructible<_Ty> && assignable_from<_Ty&, _Ty> && swappable<_Ty>;
+concept _Boolean_testable_impl = convertible_to<_Ty, bool>;
 
-// CONCEPT boolean
-#if _HAS_STD_BOOLEAN
-#define _STL_BOOLEAN_CONCEPT boolean
-#else
-#define _STL_BOOLEAN_CONCEPT _Boolean
-#endif
 template <class _Ty>
-concept _STL_BOOLEAN_CONCEPT = movable<remove_cvref_t<_Ty>>
-    && requires(const remove_reference_t<_Ty>& __x, const remove_reference_t<_Ty>& __y, const bool __b) {
-        { __x } -> convertible_to<bool>;
-        { !__x } -> convertible_to<bool>;
-        { __x && __y } -> same_as<bool>;
-        { __x && __b } -> same_as<bool>;
-        { __b && __y } -> same_as<bool>;
-        { __x || __y } -> same_as<bool>;
-        { __x || __b } -> same_as<bool>;
-        { __b || __y } -> same_as<bool>;
-        { __x == __y } -> convertible_to<bool>;
-        { __x == __b } -> convertible_to<bool>;
-        { __b == __y } -> convertible_to<bool>;
-        { __x != __y } -> convertible_to<bool>;
-        { __x != __b } -> convertible_to<bool>;
-        { __b != __y } -> convertible_to<bool>;
-    };
+concept _Boolean_testable = _Boolean_testable_impl<_Ty> && requires (_Ty&& __t) {

```

Comment:
I generally format requires-expressions on a single line if the body contains a single simple-requirement and its expression is "simple" (`requires(const foo x) { x + 1; }`). Otherwise, I layout each requirement on its own line with all requirements indented once relative to the indentation of the line that contains the parameter-declaration clause. That can result in:
```c++
template // ...
concept C = stuff && requires(T x, U y) {
    x + x;
    { y->meow() } -> D;
};
```
when the parameters appear on the first line of a concept definition, or:
```c++
template // ...
concept C = stuff && things<with, some, arguments>
    && requires(T x, U y) {
        x + x;
        { y->meow() } -> D;
    };
```
when the parameters appear on a continuation line.